### PR TITLE
Add generic message handling

### PR DESCRIPTION
**Pull request by Oluwawunmi**
### What

Add generic message handling for search-content-updated
[Ticket Link](https://jira.ons.gov.uk/browse/DIS-1665)

### How to review

Check code is ok.
Ensure all tests pass.

### Who can review

Anyone.

### CODE REVIEW

File: schema/schema.go [hunk 1]
```diff
@@ -82,3 +82,45 @@ var searchDataImport = `{
 var SearchDataImportEvent = &avro.Schema{
 	Definition: searchDataImport,
 }
+
+var releaseSchema = `{
+  "type": "record",
+  "name": "release",
```
Comment (lindenmckenzie) [inline schema/schema.go:88]:
```suggestion
  "name": "Release",
```

**Review by lindenmckenzie — CHANGES_REQUESTED** [review]

File: features/publish_data_dataset.feature [hunk 2]
```diff
@@ -1,6 +1,6 @@
 Feature: Data extractor should listen to the relevant topic and publish extracted data for Dataset API datasets
 
-
+  @Dataset
```
Comment (lindenmckenzie) [inline features/publish_data_dataset.feature:3]: Think you can do this at the Feature level.

File: features/publish_data_generic.feature [hunk 3]
```diff
@@ -0,0 +1,65 @@
+Feature:Data extractor should listen to the relevant topic and publish extracted data for Generic Upstream Services
+
+  @Generic
+  Scenario: When a valid `search-content-updated` of type release event is consumed, a `search-data-import` event is published
+    Given dp-dataset-api is healthy
+    And zebedee is healthy
+
```
Comment (lindenmckenzie) [inline features/publish_data_generic.feature:7]: Don't like the blank lines.

File: features/publish_data_generic.feature [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    When the service starts
+    And this "search-content-updated" event is queued, to be consumed
+    """
+    {
+      "URI":           "/some/uri",
+      "Title":         "title",
+      "Edition":       "something",
+      "ContentType":   "article",
+      "CDID":  "123",
```
Comment (lindenmckenzie) [inline features/publish_data_generic.feature:48]: Indentation.

File: features/steps/steps.go [hunk 5]
```diff
@@ -147,15 +174,15 @@ func (c *Component) noEventsAreProduced() error {
...
+// func arrayParser(raw string) (interface{}, error) {
+//	// remove the starting and trailing brackets
+//	str := strings.Trim(raw, "[]")
+//	if str == "" {
+//		return []string{}, nil
+//	}
+//
+//	strArray := strings.Split(str, ",")
+//	return strArray, nil
+// }
```
Comment (lindenmckenzie) [inline features/steps/steps.go:188]: Why?

File: service/service.go [hunk 6]
```diff
@@ -140,30 +156,27 @@ func (svc *Service) Close(ctx context.Context) error {
...
-			log.Info(ctx, "stopping kafka consumer listener...")
-			if err := svc.Consumer.StopAndWait(); err != nil {
-				log.Error(ctx, "error stopping kafka consumer listener", err)
-				hasShutdownError = true
-			} else {
-				log.Info(ctx, "stopped kafka consumer listener")
-			}
+		// Shutdown consumers and producer
+		if err := svc.shutdownConsumers(ctx); err != nil {
+			log.Error(ctx, "consumer shutdown errors", err)
```
Comment (lindenmckenzie) [inline service/service.go:173]:
```suggestion
			log.Error(ctx, "consumer shutdown error", err)
```

File: service/service.go [hunk 7]
```diff
@@ -219,6 +208,57 @@ func (svc *Service) Close(ctx context.Context) error {
...
+// shutdownConsumers stops and closes Kafka consumers
+func (svc *Service) shutdownConsumers(ctx context.Context) error {
+	var errMessages []string
+
+	stopAndCloseConsumer := func(consumer kafka.IConsumerGroup, name string) {
+		if consumer == nil {
+			return
+		}
+
+		log.Info(ctx, fmt.Sprintf("stopping %s kafka consumer listener...", name))
```
Comment (lindenmckenzie) [inline service/service.go:220]: No dynamic log lines.

Comment (lindenmckenzie) [reply service/service.go:220]: Repeated across the below.

**Review by lindenmckenzie — CHANGES_REQUESTED** [review]

File: features/steps/steps.go (hunk 5 above)
Comment (Oluwawunmi) [reply features/steps/steps.go]: This has now been removed entirely

Was used for reading the godog table data but we've now switched to using a json.

**Review by lindenmckenzie — APPROVED** [review]

**Pull request merged**